lib/fog/vsphere/requests/compute/get_vm_first_nvme_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Fog
+  module Vsphere
+    class Compute
+      class Real
+        def get_vm_first_nvme_controller(vm_id)
+          # require 'pry'
+          # binding.pry
+          ctrl = get_vm_ref(vm_id).config.hardware.device.find { |device| device.is_a?(RbVmomi::VIM::VirtualNVMEController) }
+          raise(Fog::Vsphere::Compute::NotFound) unless ctrl
```

Comment:
I'd be explicit what wasn't found
```suggestion
          raise Fog::Vsphere::Compute::NotFound, "No NVME controller found for #{vm_id}" unless ctrl
```